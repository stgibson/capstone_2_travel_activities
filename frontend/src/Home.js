import React, { useState } from "react";
import ActivitySearch from "./ActivitySearch";
import ActivityList from "./ActivityList";

/**
 * Component that allows users to search for activities based on location
 * @param {Object} param0 
 * @returns JSX code for rendering home page
 */
const Home = ({ getActivitiesByLocation }) => {
  const [activities, setActivities] = useState([]);

  const searchActivities = async data => {
    const activitiesResults = await getActivitiesByLocation(data);
    if (activitiesResults) {
      setActivities(activitiesResults);
    }
  };

  return (
    <>
      <div>
        <ActivitySearch searchActivities={ searchActivities } />
      </div>
      <br />
      <div>
        <ActivityList
          title="Results:"
          activities={ activities }
        />
      </div>
    </>
  );
};

export default Home;