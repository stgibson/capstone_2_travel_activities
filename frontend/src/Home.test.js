import React from "react";
import { MemoryRouter } from "react-router-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import Home from "./Home";
import rootReducer from "./reducers/root";

const store = createStore(rootReducer);

const getActivitiesByLocation = jest.fn();

it("renders without crashing", () => {
  render(
    <Provider store={ store }>
      <MemoryRouter>
        <Home getActivitiesByLocation={ getActivitiesByLocation } />
      </MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={ store }>
      <MemoryRouter>
        <Home getActivitiesByLocation={ getActivitiesByLocation } />
      </MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
