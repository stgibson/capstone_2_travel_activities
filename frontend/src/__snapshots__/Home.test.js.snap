// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Search for Activities
    </h2>
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="city"
        >
          City
        </label>
        <input
          class="form-control"
          id="city"
          type="text"
          value="paris"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="country"
        >
          Country
        </label>
        <input
          class="form-control"
          id="country"
          type="text"
          value="france"
        />
      </div>
    </form>
    <button
      class="btn btn-primary"
      type="button"
    >
      Search
    </button>
  </div>
  <br />
  <div>
    <div>
      <h2>
        Results:
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
